### PR TITLE
i70_continuous_integration

### PR DESCRIPTION
**Pull request by jamiebull1**
I've taken another look at this and at what some other projects are doing to handle cross-platform builds. Mac is the missing link, but both Linux and Windows have CI providers that are free for open source projects.

This PR includes CI setup for Linux (Travis) and Windows (32 and 64 bit with Appveyor), all tested against Python 2.7.x. Code coverage is checked using CodeCov, based on the Travis build. All three of these services are really simple to set up for a repo. The hard part was getting the two .yml files right but that should transfer over without needing any changes.

There's one change that does need making, since the badges in `README.md` are repo- and branch-specific. At the moment I have them referencing `jamiebull1/eppy/i70_continuous_integration` for the sake of showing them passing tests but they should eventually reference `santoshphilip/eppy/develop`, `santoshphilip/eppy/master` or whatever. It's unfortunately not possible to automate those references so they need changing for each branch.

You can see the shiny badges on the `i70_continuous_integration` README.md [here](https://github.com/jamiebull1/eppy/tree/i70_continuous_integration).

### CODE REVIEW

**Comment by jamiebull1** [comment]
@santoshphilip This is really helping me at the moment with catching python 3 problems. It would be great to have in the develop branch if you have the time (and the inclination) to:

Authorise Travis to run tests on Linux (runs tests for 2.7, runs 2to3 and tests for 3.5):

 - Go [here](https://travis-ci.org/auth) and login with GitHub
 - Add `santoshphilip/eppy`

Authorise Appveyor to run tests on Windows (runs tests for 2.7, runs 2to3 and tests for 3.5):

- Sign in with GitHub [here](https://ci.appveyor.com/login)
 - Add `santoshphilip/eppy` (actually it looks like I can do this as a contributor if that's easier)

Authorise CodeCov (measures test coverage):

 - Sign in with GitHub [here](https://codecov.io/)
 - Add `santoshphilip/eppy`

Let me know if you want any changes to the README, or any other thoughts you have on this.

**Comment by santoshphilip** [comment]
I shall try it out

**Comment by santoshphilip** [comment]
This stuff is looking good. I just took a quick look. Let me to play with it for a few days.

File: .gitignore [hunk 1]
```diff
@@ -9,3 +9,4 @@ __pycache__
 .pydevproject
 .settings/
 /remove_licence.py
+/.coverage
```
Comment (jamiebull1) [inline .gitignore]: This stops the coverage report polluting the repo

File: README.md [hunk 2]
```diff
@@ -1,7 +1,18 @@
 Eppy
 ====
+[![PyPI](https://img.shields.io/pypi/dm/eppy.svg)](https://pypi.python.org/pypi/eppy)
+ from PyPI
 
-Eppy is a scripting language for EnergyPlus idf files, and EnergyPlus output files. Eppy is written in the programming language Python. As a result it takes full advantage of the rich data structure and idioms that are avaliable in python. You can programmatically navigate, search, and modify EnergyPlus idf files using eppy. The power of using a scripting language allows you to do the following:
+[![Travis](https://img.shields.io/travis/jamiebull1/eppy/i70_continuous_integration.svg)](https://travis-ci.org/jamiebull1/eppy)
```
Comment (jamiebull1) [inline README.md]: change "jamiebull1" to "santoshphilip" when merged

File: README.md [hunk 3]
```diff
@@ -1,7 +1,18 @@
...
+ for Python 2.7 and 3.5 on Linux via Travis
+
+[![Appveyor](https://img.shields.io/appveyor/ci/jamiebull1/eppy/i70_continuous_integration.svg)](https://ci.appveyor.com/api/projects/status/github/jamiebull1/eppy)
```
Comment (jamiebull1) [inline README.md]: change "jamiebull1" to "santoshphilip" when merged

File: README.md [hunk 4]
```diff
@@ -1,7 +1,18 @@
...
+ for Python 2.7 and 3.5 on Windows via Appveyor
+
+[![CodeCov](https://img.shields.io/codecov/c/github/jamiebull1/eppy/i70_continuous_integration.svg)](https://codecov.io/github/jamiebull1/eppy)
```
Comment (jamiebull1) [inline README.md]: change "jamiebull1" to "santoshphilip" when merged

File: appveyor.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
+environment:
+
+  matrix:
+
+    - PYTHON: "C:\\Python27"
+    - PYTHON: "C:\\Python27-x64"
+    - PYTHON: "C:\\Python35"
```
Comment (jamiebull1) [inline appveyor.yml]: Add Python35-x64

**Comment by jamiebull1** [comment]
@santoshphilip I've added these comments as an example of how we could be using the PR system for code review.

**Comment by jamiebull1** [comment]
@santoshphilip Have you had a chance to look at this properly yet? It's definitely top of my wish-list for the develop branch.

**Comment by santoshphilip** [comment]
@jamiebull1 
Please go ahead and take any steps to make continuous integration happen.
If you need to me take any specific steps as owner, let me know and I'll do so.

**Comment by jamiebull1** [comment]
@santoshphilip Great. If you can set up the three things in [this post](https://github.com/santoshphilip/eppy/pull/76#issuecomment-182766188) I can do the rest.

**Comment by santoshphilip** [comment]
I have authorized:

- Travis
- Appveyor
- CodeCov

from my account.
Check if all is well. See if I need to take any further steps.

**Comment by jamiebull1** [comment]
@santoshphilip Awesome, thanks. The CodeCov and Travis accounts are working but it looks like there's something not quite right with the Appveyor one. I get a "project not found or access denied" error message. I don't know if you can see the results? If you can then it's probably something to do with permissions.

**Comment by santoshphilip** [comment]
@jamiebull1 Looks like I had not completed the process on Appveyor.
it may be good now.Try it again and see.

**Comment by jamiebull1** [comment]
@santoshphilip - Yes, that's all working now. It's ready to go into the `develop` branch if you're happy with it.

**Comment by santoshphilip** [comment]
I'll merge it into Develop as soon as I get a moment.

**Pull request merged**